core/src/main/scala/chisel3/Data.scala

```diff
@@ -607,15 +607,18 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
 
   /** Require that two things are type equivalent, and if they are not, print a helpful error message as
     * to why not.
+    *
+    * @param that the Data to compare to for type equivalence
+    * @param message if they are not type equivalent, contextual message to add to the exception thrown
     */
-  private[chisel3] def requireTypeEquivalent(that: Data): Unit = {
+  def requireTypeEquivalent(that: Data, message: String = ""): Unit = {
```

Comment:
The other similar APIs are standalone object with apply methods in package `chisel3.experimental` (see [requireIsAnnotatable](https://www.chisel-lang.org/api/latest/chisel3/experimental/package$$requireIsAnnotatable$.html), [requireIsChiselType](https://www.chisel-lang.org/api/latest/chisel3/experimental/package$$requireIsChiselType$.html), [requireIsHardware](https://www.chisel-lang.org/api/latest/chisel3/experimental/package$$requireIsHardware$.html)). Granted, each of those only takes a single argument, but I'm wondering if this should be consistent with them rather than being a method on Data?

Alternatively, the existing check APIs are on `DataMirror` ([checkTypeEquivalence](https://www.chisel-lang.org/api/latest/chisel3/reflect/DataMirror$.html#checkTypeEquivalence(x:chisel3.Data,y:chisel3.Data):Boolean), and [checkAlignmentTypeEquivalence](https://www.chisel-lang.org/api/latest/chisel3/reflect/DataMirror$.html#checkAlignmentTypeEquivalence(x:chisel3.Data,y:chisel3.Data):Boolean).

I'm not sure what we should do but I think I lean toward the `DataMirror` option, WDYT?

Reply:
yeah i think this should be like data mirror

Reply:
moved to DataMirror.requireTypeEquivalent